### ISSUE TITLE
functional test migration follow-up issues

### ISSUE THREAD

**Issue by TianyuZuo** [eclipse/openj9#1708]
functional tests will be moved to `functional` folder in `openj9/test/`, Issue #1679  once this movement is done. We need some follow up PR to enhance this movement.

1. move `getdependency to functional level. `
2. make build_list more flexible - it can be target to each sub level of test folder.
3. update readme file to pointing to the right position and mention functional folder.

**Comment by TianyuZuo** [comment]
fyi @llxia  @renfeiw

**Comment by smlambert** [comment]
I believe all the items relating to this issue have been addressed, so will close it.  We can reopen or create a new issue if more items arise.
